Tests/CSharp/StatementTests.cs

```diff
@@ -1403,7 +1403,7 @@ public bool CanDoWork(object Something)
     {
         switch (true)
         {
-            case object _ when (DateAndTime.Today.DayOfWeek == DayOfWeek.Saturday) | (DateAndTime.Today.DayOfWeek == DayOfWeek.Sunday):
+            case object _ when ((int)DateAndTime.Today.DayOfWeek == (int)DayOfWeek.Saturday) | ((int)DateAndTime.Today.DayOfWeek == (int)DayOfWeek.Sunday):
```

Comment:
These are both enums of the same type so I'm surprised converting them was necessary

Reply:
The difficulty is we handle enum<->int in the type conversion analysis, rather than in the binary expression visitor. `typeInfo.Type` is the enum type, and `typeInfo.ConvertedType` is int, so we insert the cast (which is required everywhere else). This makes sense, since presumably VB just does an integer comparison for enums. Fine for the compiler, kind of ugly for converted code.

There's a few ways we could handle this, I haven't put too much thought into it though.